File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    prog = deque(progresses)
    spd = deque(speeds)
    answer = []
    
    #for progress, speed in zip(progresses, speeds):
    while prog:
        for i in range(len(prog)):
            prog[i] += spd[i]
        
        cnt = 0
        while prog and prog[0] >= 100:
            prog.popleft()
            spd.popleft()
            cnt += 1
        
        if cnt:
            answer.append(cnt)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int tday = ceil((float)(100-progresses[0])/speeds[0]), day;
    int cnt = 1;
    
    for (int i=1; i<progresses.size(); i++){
        day = ceil((float)(100-progresses[i])/speeds[i]);
        if (tday >= day)
            cnt++;
        else {
            answer.push_back(cnt);
            tday = day;
            cnt = 1;
        }
    }
    answer.push_back(cnt);
    
    return answer;
}